pkg/openthread/contrib/platform_uart.c

```diff
@@ -13,34 +13,163 @@
+            gSerialMessage[currentSerialBufferNumber]->buf[frameLength] = (uint8_t) c;
+            gSerialMessage[currentSerialBufferNumber]->serial_buffer_status = OPENTHREAD_SERIAL_BUFFER_STATUS_READY_TO_PROCESS;
+            gSerialMessage[currentSerialBufferNumber]->length = frameLength + 1;
+            msg.content.ptr = gSerialMessage[currentSerialBufferNumber];
+            msg_send_int(&msg, openthread_get_pid());
+        }
+        else {
+            gSerialMessage[currentSerialBufferNumber]->serial_buffer_status = OPENTHREAD_SERIAL_BUFFER_STATUS_FREE;
+        }
+        gOnGoingSpinelReception = false;
+        frameLength = 0;
+    }
+    else if (gOnGoingSpinelReception == true) {         /* Payload of Spinel Frame */
+        if ( OPENTHREAD_ERROR_NO_EMPTY_SERIAL_BUFFER == currentSerialBufferNumber ) {
+            return;
+        }
+        if ( OPENTHREAD_SERIAL_BUFFER_STATUS_FULL != gSerialMessage[currentSerialBufferNumber]->serial_buffer_status) {
+            gSerialMessage[currentSerialBufferNumber]->buf[frameLength] = (uint8_t) c;
+        }
+    }
+
+    if (gOnGoingSpinelReception == true) {
+        frameLength++;
+        if ( frameLength >= OPENTHREAD_SERIAL_BUFFER__PAYLOAD_SIZE) {
+            DEBUG("SERIAL: ERROR => OPENTHREAD_SERIAL_BUFFER__PAYLOAD_SIZE overflowed\n");
+            gSerialMessage[currentSerialBufferNumber]->serial_buffer_status = OPENTHREAD_SERIAL_BUFFER_STATUS_FULL;
+        }
+    }
+}
+
+#else
+
+void uart_handler(void* arg, char c) {
+    static uint16_t frameLength = 0;
+    if (frameLength == 0 && gSerialMessage != NULL) {
+        memset(gSerialMessage[0], 0, sizeof(serial_msg_t));
+    }
+    switch (c) {
+        case '\r':
+        case '\n':
+            if (frameLength > 0) {
+                gSerialMessage[0]->buf[frameLength] = c;
+                frameLength++;
+                gSerialMessage[0]->length = frameLength;
+                msg_t msg;
+                msg.type = OPENTHREAD_SERIAL_MSG_TYPE_EVENT;
+                msg.content.ptr = gSerialMessage[0];
+                msg_send_int(&msg, openthread_get_pid());
+                frameLength = 0;
+            }
+            break;
+        default:
+            if (frameLength < OPENTHREAD_SERIAL_BUFFER_SIZE) {
+                gSerialMessage[0]->buf[frameLength] = c;
+                frameLength++;
+            }
+            break;
+    }
+}
+
+#endif /* MODULE_OPENTHREAD_NCP_FTD */
 
 /* OpenThread will call this for enabling UART (required for OpenThread's CLI)*/
-ThreadError otPlatUartEnable(void)
+otError otPlatUartEnable(void)
 {
-    return kThreadError_None;
+    for (uint8_t i = 0; i < OPENTHREAD_NUMBER_OF_SERIAL_BUFFER; i++) {
+        gSerialMessage[i] = (serial_msg_t*) &gSerialBuff[i];
+        gSerialMessage[i]->serial_buffer_status = OPENTHREAD_SERIAL_BUFFER_STATUS_FREE;
+    }
+
+    uart_init(OPENTHREAD_UART_DEV, OPENTHREAD_UART_BAUDRATE, (uart_rx_cb_t) uart_handler, NULL);
```

Comment:
And here use UART_STDIO_DEV and UART_STDIO_BAUDRATE. Then you use the value defined in the board config and there's no need for passing the baudrate in the application makefile.